### DISCUSSION TITLE
Table sortable on a custom field?

### DISCUSSION THREAD

**Discussion by RPotkowaVegasystems** [filamentphp/filament#9321] [General]
I have a custom Textfield, where i sum a field from models that are related? How it is possible  also to sort this field on the values?

```
 TextColumn::make('custom_sum_field')
                    ->label(__('Size Bytes'))
                    ->getStateUsing(function ($record) {
                        return $record->hasManyRelationship->sum('my_field');
                    }),
```

**Comment by sinnbeck** [comment]
You can pass a query to `->sortable()`

https://filamentphp.com/docs/3.x/tables/columns/getting-started#sorting Look at the last example
